src/main/java/ee/tuleva/onboarding/mandate/email/MandateEmailService.java

```diff
@@ -21,57 +25,90 @@
 public class MandateEmailService {
   private final EmailService emailService;
   private final MandateEmailContentService emailContentService;
+  private final Clock clock;
 
   public void sendMandate(
       User user,
       Mandate mandate,
       PillarSuggestion pillarSuggestion,
       UserPreferences contactDetails,
       Locale locale) {
+    switch (pillarSuggestion.getOtherPillar()) {
+      case 2 -> sendSecondPillarEmail(user, mandate, pillarSuggestion, locale);
+      case 3 -> sendThirdPillarEmails(user, mandate, pillarSuggestion, contactDetails, locale);
+      default -> throw new IllegalArgumentException(
+          "Unknown pillar: " + pillarSuggestion.getOtherPillar());
+    }
+  }
```

Comment:
Any way to solve this polymorphically instead of branching? We should avoid logic like 
`if 2nd pillar then do this, else if 3rd pillar then do that, else throw exception` at _all costs_. Ideally, we would do it in 1 place only and then use polymorphism in the rest of the codebase.